src/Temporalio/Converters/EncodedRawValue.cs

```diff
@@ -0,0 +1,20 @@
+using System;
+using System.Threading.Tasks;
+using Temporalio.Api.Common.V1;
+
+namespace Temporalio.Converters
+{
+    /// <summary>
+    /// Implementation of a <see cref="IEncodedRawValue" />.
+    /// </summary>
+    /// <param name="DataConverter">Data converter to perform conversions.</param>
+    /// <param name="Payload">Raw payload.</param>
+    public record EncodedRawValue(DataConverter DataConverter, Payload Payload) : IEncodedRawValue
```

Comment:
Is this a .NET thing that everything needs both an interface and a class? Is there other reasoning behind this?

Reply:
I need an implementation of the interface somewhere, otherwise how do I use it? I do think there is technically value in making this implementation public because you can technically cast and get the payload and data converter back out. But technically I could make this `internal`. But yeah, I have to have an implementation because this is what I set as the `Memo` dictionary keys for workflow executions coming back from client calls.

Reply:
Just wondering why you need both instead of just the implementation.

Reply:
There was a request to hide payloads from users like https://pkg.go.dev/go.temporal.io/sdk/converter#EncodedValue does. If we're not gonna do that, why not just use payloads for these things and make the converter available for use (an option brought up on internal Slack, but this was preferred).